src/models/palindrome/palindrome.go

```diff
@@ -0,0 +1,52 @@
+package palindrome
+
+import (
+	"errors"
+	"strings"
+)
+
+// SubPalindromes returns a string of all subpalindromes of its argument
+func SubPalindromes(stringToExplore string) (resultString string, err error) {
+	stringToExplore = strings.Replace(stringToExplore, " ", "", -1)
+	var palindromeSubstrings [][]rune
+	err=valid(stringToExplore)
+	if err!= nil{
+		return stringToExplore, err
+	}
+	length := len([]rune(stringToExplore))
+	runedStr := []rune(stringToExplore)
+	for g, _ := range stringToExplore {
+		for substringLength := 1; substringLength <= length-g; substringLength++ {
+			substring := runedStr[g : g+substringLength]
+			if isPalindrome(substring) && len(substring) > 1 {
+				palindromeSubstrings = append(palindromeSubstrings, substring)
+			}
+		}
+	}
+	for j := range palindromeSubstrings {
+		for i := range palindromeSubstrings[j] {
+			resultString += string(palindromeSubstrings[j][i])
+		}
+		resultString += " "
+	}
+	return
+}
+
+func isPalindrome(stringToTest []rune) bool {
```

Comment:
I think it would be better to pass `stringToTest` to this function as a `string` and then convert it to `[]rune` inside the function.

Reply:
I was trying to do this, but it only brings mess into my code (maybe i just don't understand smth right).  Because in my function "SubPalindromes" i need to calculate offset for checking if substring of "stringToExplore" is a palindrome. And this offset need to be checked in runes, cause of the different size of the characters. So it comes up that i need to call "IsPalindrome" function several times just to get my string variable in runes for calculating offset and to convert my variable "substring" to string to call the "IsPalindrome" function.  
![mess](https://user-images.githubusercontent.com/32939931/39997816-d6ae9f6c-578c-11e8-8c59-8612be3c33e9.JPG)